homeassistant/components/microbees/coordinator.py

```diff
@@ -53,15 +53,10 @@ async def _async_update_data(self) -> MicroBeesCoordinatorData:
             except MicroBeesException as err:
                 raise UpdateFailed(f"Error communicating with API: {err}") from err
 
-            bees_dict = {}
-            actuators_dict = {}
-            sensors_dict = {}
-            for bee in bees:
-                bees_dict[bee.id] = bee
-                for actuator in bee.actuators:
-                    actuators_dict[actuator.id] = actuator
-                for sensor in bee.sensors:
-                    sensors_dict[sensor.id] = sensor
             return MicroBeesCoordinatorData(
-                bees=bees_dict, actuators=actuators_dict, sensors=sensors_dict
+                bees={bee.id: bee for bee in bees},
```

Comment:
It looks like it has to iterate bees more times now?

Reply:
can do this instead:

```
actuators = {}
sensors = {}

for bee in bees:
    actuators.update({actuator.id: actuator for actuator in bee.actuators})
    sensors.update({sensor.id: sensor for sensor in bee.sensors})

return MicroBeesCoordinatorData(
            bees={bee.id: bee for bee in bees},
            actuators=actuators,
            sensors=sensors,
        )
```

Not too much difference in efficiency is bees is small.

Reply:
I guess you could build the bees_id_mapping in the first loop as well.. but the dict comp might even be faster with iterating it twice.  Probably need to use `timeit` to tell

Reply:
Best to time it with the actual device for sure

Reply:
Changed the title.

Leave the coordinator as is, or revert to one long list/dict comprehension?

Reply:
I'd probably revert the change in the coordinator as it makes the PR mergable right away

Reply:
I was thinking you were going to revert the change to coordinator.py and work with the code owner to profile this  in a future PR as the other changes are mergable

Reply:
Oh, back to the original?  I can do that.

Reply:
done...